### DISCUSSION TITLE
Session's SessionData typescript types should be able to be strongly-typed without `any`

### DISCUSSION THREAD

**Discussion by NateRadebaugh** [remix-run/remix#3025] [Proposals]
`@remix-run/server-runtime/sessions.d.ts`

Currently, there's no supported way to enforce that Session's SessionData typescript type is strongly-typed and is instead a key-value record.

I think ideally `createCookieSessionStorage()` would take a generic to specify the type of session data for read/writes:

```ts
const { getSession } = createCookieSessionStorage<UserDetails>();
```

**Comment by magnusdr** [comment]
If you want to achieve this, you could create a file `types/@remix-run/node.d.ts` like this
```ts
import "@remix-run/node";

declare module "@remix-run/node" {
  interface SessionData {
    preferredLocale: string; // as an example
  }
}
```
